r-rse/course-design.md

```diff
@@ -0,0 +1,135 @@
+# Course design components for *RSE in R*
+
+## Basic design components
+
+These are to develop a compass for developing and structuring the material.
+Many may have a strong focus on "in class" etc, 
+but this is to give a direction on how to write those sections
+(e.g. having the perspective that an instructor will be reading these sections).
+
+- **Learning outcome/goal**: Ultimately, to be able to create an R package, that:
+    - Passes checks and requirements to (potentially) be submitted to CRAN
+    - Incorporates best practices in package development 
+    (unit testing, continuous integration, version control)
+    - Is designed to involve the community, such as gaining potential contributors
+    (including licensing, code of conduct, contributing guidelines, 
+    well described README)
+    - Has user-friendly (i.e. targetted to a novice to intermediate user)
+    documentation and tutorials on how to use the package (e.g. a nice website)
+    - Is adequately structured to make package management and maintanence easier
+    (e.g. with Issue Templates, Project Boards, Labels, contributing guidelines)
+    
+- **Teaching/learning activities**:
+    - In class portion: 
+        - Some lectures for those material that aren't coding heavily
+        (e.g. for lessons to involve community, being a good community member)
+        - Live coding (with instructor)
+        - Exercises that expand on material covered with instructor that 
+        solve a provided problem, with or without scaffolding
+    - Out of class: 
+        - Group work/project creating an R package based on a set of (given?) problems
+        - Formative assessments like coding with immediate feedback on correct/incorrect
+        (via learnr package? Esp with new update to RStudio in version 1.3)
+        - Reading assignments?
+
+- **Assessment and feedback activities**:
+    - Deadlines for milestones to acheive for project, with instructor feedback on progress
+    - Out-of-class formative assessments based on maybe learnr tutorials?
+    - After completing in-class exercises, go over them as class?
+
+- Key resources to use for this:
+  - R packages: usethis, devtools, drake?, pkgdown, testthat, styler, rmarkdown
+  - Online books: [R Packages](https://r-pkgs.org/) 
+  (we differentiate from this amazing book by structuring everything around 
+  this being a course, with exercises, assessments, (group?) projects, 
+  instructions/comments/notes for instructors, etc)
```

Comment:
Right, I think of R packages as being a reference book, we are writing for people learning this for the first time.